app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -145,13 +145,19 @@ def remove_from_environment
     param :system_environment_id, :number, :desc => N_("environment to reassign orphaned systems to")
     param :key_content_view_id, :number, :desc => N_("content view to reassign orphaned activation keys to")
     param :key_environment_id, :number, :desc => N_("environment to reassign orphaned activation keys to")
+    param :force_destory_cv, :boolean, :desc => N_("delete the content view if all versions are also removed")
     def remove
-      cv_envs = ContentViewEnvironment.where(:environment_id => params[:environment_ids],
-                                             :content_view_id => params[:id]
-                                            )
-      versions = @content_view.versions.where(:id => params[:content_view_version_ids])
+      if params[:force_destory_cv]
+        cv_envs = ContentViewEnvironment.where(:content_view_id => params[:id])
+        versions = @content_view.versions
+      else
+        cv_envs = ContentViewEnvironment.where(:environment_id => params[:environment_ids],
+                                               :content_view_id => params[:id]
+        )
+        versions = @content_view.versions.where(:id => params[:content_view_version_ids])
+      end
 
-      if cv_envs.empty? && versions.empty?
+      if !params[:force_destory_cv] && cv_envs.empty? && versions.empty?
         fail _("There either were no environments nor versions specified or there were invalid environments/versions specified. "\
                "Please check environment_ids and content_view_version_ids parameters.")
       end
```

Comment:
what does it mean to force destroy a cv that has activation keys & hosts assigned? what happens to them?

Reply:
nevermind, i misunderstood what this was checking :)